### ISSUE TITLE
is there something wrong with hard negative mining?

### ISSUE THREAD

**Issue by KaleidoZhouYN** [XiaXuehai/faceboxes#16]
the implement of hard negative mining actually decrease the anchor number so when positive number = 0,negative number will be = 0 too

**Comment by KaleidoZhouYN** [comment]
> positive number != 0, because https://github.com/XiaXuehai/faceboxes/blob/master/dataset.py#L131.
> There is an imbalance between the positive and negative examples if you do not use hard negative mining.

yes,the box in the random crop will no be 0 but the positive anchor will be 0 because of the IOU

**Comment by XiaXuehai** [comment]
When encoding, we set the max IOU to be the positive anchor.
